### PR TITLE
feat(authz): Add lib with entity representation flattening

### PR DESCRIPTION
**Pull request by elizabethhealy**
resolves https://github.com/opentdf/platform/issues/644
breaking https://github.com/opentdf/platform/issues/645 into separate PR

### CODE REVIEW

File: service/internal/subjectmappingbuiltin/subject_mapping_builtin.go [hunk 1]
```diff
@@ -187,7 +188,11 @@ ConditionEval:
 }
 
 func EvaluateCondition(condition *policy.Condition, entity map[string]any) (bool, error) {
-	mappedValues, err := ExecuteQuery(entity, condition.GetSubjectExternalSelectorValue())
+	flattenedEntity, err := flattening.Flatten(entity)
+	if err != nil {
+		return false, err
```
Comment (jakedoublev) [inline service/internal/subjectmappingbuiltin/subject_mapping_builtin.go:193]: Can we wrap these errors? 
```suggestion
		return false, fmt.Errorf("failure to EvaluateCondition in subject mapping builtin: %w", err)
```

**Review by jakedoublev — DISMISSED** [review]

File: service/internal/subjectmappingbuiltin/subject_mapping_builtin.go (hunk 1 above)
Comment (elizabethhealy) [reply service/internal/subjectmappingbuiltin/subject_mapping_builtin.go]: will do in second PR

File: lib/flattening/go.mod [hunk 2]
```diff
@@ -1,6 +1,6 @@
 module github.com/opentdf/platform/lib/flattening
 
-go 1.22.4
+go 1.21
```
Comment (jakedoublev) [inline lib/flattening/go.mod:3]: Was this downgrade intentional?

Comment (jakedoublev) [reply lib/flattening/go.mod:3]: Hmm, my diff view got out of sync for some reason. Disregard.

**Review by jakedoublev — APPROVED** [review]
Not sure just by looking at this if the pipeline is perfectly updated but these changes look good to me.

**Review by strantalis — APPROVED** [review]

**Pull request merged**